cmd/config/config.go

```diff
@@ -33,18 +46,22 @@ func NewConfigCmd(analyticsTracker analytics.Tracker) *cobra.Command {
 		PreRun: func(cmd *cobra.Command, args []string) {
 			// only track event if there are flags
 			if len(os.Args[1:]) > 1 {
-				analyticsTracker.SendCommandRunEvent(
+				analyticsTrackerFn(
```

Comment:
Since we don't get the flag values until we run the command, we can pass in the factory function in the command's constructor and call it when we have the flag values. The factory function already has an ID, so additional calls from the same factory function will all have the same ID for correlation.